ios/RNPhotosFramework/PHAssetsService.m

```diff
@@ -93,19 +93,35 @@ +(NSMutableDictionary *)extendAssetDicWithAssetMetaData:(NSMutableDictionary *)d
     return dictToExtend;
 }
 
-+(NSMutableArray<PHAsset *> *) getAssetsForFetchResult:(PHFetchResult *)assetsFetchResult startIndex:(NSUInteger)startIndex endIndex:(NSUInteger)endIndex {
++(NSMutableArray<PHAsset *> *) getAssetsForFetchResult:(PHFetchResult *)assetsFetchResult startIndex:(int)startIndex endIndex:(int)endIndex andReverseIndices:(BOOL)reverseIndices {
     
     NSMutableArray<PHAsset *> *assets = [NSMutableArray new];
-    [assetsFetchResult enumerateObjectsUsingBlock:^(PHAsset *asset, NSUInteger index, BOOL *stop) {
-        if(index >= startIndex){
-            [assets addObject:asset];
+    int assetCount = assetsFetchResult.count;
+    if(assetCount != 0) {
+        if(reverseIndices) {
+            int originalStartIndex = startIndex;
+            startIndex = assetCount - endIndex;
+            endIndex = assetCount - originalStartIndex;
         }
-        if(index >= endIndex){
-            *stop = YES;
-            return;
+        if(startIndex < 0) {
+            startIndex = 0;
         }
-        
-    }];
+        if(endIndex < 0) {
+            endIndex = 0;
+        }
+        if(startIndex >= assetCount) {
+            startIndex = assetCount -1;
+        }
+        if(endIndex >= assetCount) {
+            endIndex = assetCount -1;
+        }
+        NSIndexSet *indexSet = [NSIndexSet indexSetWithIndexesInRange:NSMakeRange(startIndex, endIndex - startIndex)];
+        NSEnumerationOptions enumerationOptions = reverseIndices ? NSEnumerationConcurrent : NSEnumerationReverse;
```

Comment:
@justinsherwood This must be wrong. Change it in your branch to         `NSEnumerationOptions enumerationOptions = reverseIndices ? NSEnumerationReverse : NSEnumerationConcurrent;`